File: nysol_python/mcmd/methods/convTex/db/smp/func_regexm.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='regexm'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,caabaa
2,acabaaa
3,
4,bbcbcc
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``id=1,id=2`` 共に、正規表現で示された ``c`` に続く ``aa`` を含んでいるが、
``id=2`` ではマッチする範囲が部分的(2文字目の ``c`` から最後まで)であるために偽となっている。
'''
script['sh_code']='''
mcal c='regexm($s{str},"c.*aa")' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='regexm($s{str},"c.*aa")', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='末尾一致'
script['comment']='''
正規表現 ``.*c`` で$str$項目の全体がカバーされるのは ``id=4`` の行のみである。
'''
script['sh_code']='''
mcal c='regexm($s{str},".*c")' a=rsl i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mcal(c='regexm($s{str},".*c")', a='rsl', i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='空文字マッチ'
script['comment']='''
正規表現 ``^$`` で ``id=3`` の空文字にマッチする。
'''
script['sh_code']='''
mcal c='regexm($s{str},"^$")' a=rsl i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcal(c='regexm($s{str},"^$")', a='rsl', i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_regexm.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,caabaa
2,acabaaa
3,
4,bbcbcc
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Both records where \verb|id=1,id=2| contains regular expression beginning with \verb|c| and ends with \verb|aa|, in record \verb|id=2|, there is only partial match (matches from \verb|c| at the second position to the end) with the regular expression.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='regexm($s{str},"c.*aa")' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="String ends with same substring"
comment=<<'EOF'
The full string in column $str$ that matches the regular expression \verb|.*c| is only found in the record where \verb|id=4|.
EOF
scp=<<'EOF'
mcal c='regexm($s{str},".*c")' a=rsl i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Match blank characters"
comment=<<'EOF'
Match the blank characters at \verb|id=3| using the regular expression \verb|^$|.
EOF
scp=<<'EOF'
mcal c='regexm($s{str},"^$")' a=rsl i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)


